src/sinks/http.rs

```diff
@@ -193,6 +197,17 @@ impl SinkConfig for HttpSinkConfig {
         let (framer, serializer) = self.encoding.build(SinkType::MessageBased)?;
         let encoder = Encoder::<Framer>::new(framer, serializer);
 
+        let (payload_prefix, payload_suffix) = match (encoder.serializer(), encoder.framer()) {
+            (
+                Serializer::Json(_),
+                Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }),
+            ) => validate_payload_wrapper(&self.payload_prefix, &self.payload_suffix)?,
+            _ => (
+                self.payload_prefix.to_owned(),
+                self.payload_suffix.to_owned(),
+            ),
+        };
```

Comment:
I think this would look a little cleaner to have `validate_payload_wrapper()` take a reference to the Encoder as an argument, and extract this match to the helper function.